src/github.ts

```diff
@@ -479,7 +510,33 @@ export class GitHub {
         sha: graphCommit.sha,
         message: graphCommit.message,
       };
-      const pullRequest = graphCommit.associatedPullRequests.nodes.find(pr => {
+
+      const authors = new Set<string>();
+      const associatedPRs = graphCommit.associatedPullRequests.nodes;
+
+      if (associatedPRs.length > 0) {
+        // Add authors from commits of associated PRs
+        associatedPRs.forEach(pr => {
+          pr.commits.nodes.forEach(node => {
+            node.commit.authors.nodes
+              .map(author => author.name)
+              .forEach(authors.add, authors);
+          });
+        });
```

Comment:
I think this will grab authors that are not necessarily intended. Associated pull requests include any PRs that are mentioned in the commit message, not necessarily the PR that created it.

So if we mention another PR by another author, this will grab that PR's author who should not necessarily be attached to the current PR.

This weird behavior is why we have that logic to find the associated pull request whose merge commit SHA matches the commit SHA (skip the mentioned PRs)

Reply:
I'm a little puzzled, and I'm a little curious about what scenario this will happen in. Can you tell me how to trigger this problem?

Reply:
If you reference another PR in the PR body like `See #1234`, then graphql will return that as an associatedPullRequest.